elasticapm/contrib/sanic/__init__.py

```diff
@@ -0,0 +1,338 @@
+        @entity.middleware("response")
+        async def _instrument_response(request: Request, response: HTTPResponse):
+            await set_context(
+                lambda: get_response_info(
+                    config=self._client.config,
+                    response=response,
+                    skip_headers=self._skip_headers,
+                ),
+                "response",
+            )
+            self._setup_transaction_name(request=request)
+            result = f"HTTP {response.status // 100}xx"
+            set_transaction_result(result=result, override=False)
+            set_transaction_outcome(http_status_code=response.status, override=False)
+            elastic_context(data={"status_code": response.status}, key="response")
+            self._client.end_transaction()
+
+    def _setup_transaction_name(self, request: Request) -> None:
+        """
+        Method used to setup the transaction name using the provided callback or the default mode
+        :param request: Incoming HTTP Request entity
+        :return: None
+        """
+        if self._transaction_name_callback:
+            name = self._transaction_name_callback(request)
+        else:
+            name = self._default_transaction_name_generator(request=request)
+        if name:
+            set_transaction_name(name, override=False)
+
+    # noinspection PyBroadException,PyProtectedMember
+    def _setup_exception_manager(self):
+        """
+        Setup global exception handler where all unhandled exception can be caught and tracked to APM server
+        :return:
+        """
+
+        # noinspection PyUnusedLocal
+        async def _handler(request: Request, exception: BaseException):
+            if not self._client:
+                return
+
+            self._client.capture_exception(
+                exc_info=sys.exc_info(),
+                context={
+                    "request": await get_request_info(
+                        config=self._client.config, request=request, skip_headers=self._skip_headers
+                    ),
+                },
+                handled=True,
+            )
+            self._setup_transaction_name(request=request)
+            set_transaction_result(result="HTTP 5xx", override=False)
+            set_transaction_outcome(outcome=constants.OUTCOME.FAILURE, override=False)
+            elastic_context(data={"status_code": 500}, key="response")
+            self._client.end_transaction()
+
+        if not isinstance(self._app.error_handler, ElasticAPMPatchedErrorHandler):
+            patched_client = ElasticAPMPatchedErrorHandler()
+            patched_client.setup_apm_handler(apm_handler=_handler)
+            patched_client.handlers = self._app.error_handler.handlers
+            patched_client.cached_handlers = self._app.error_handler.cached_handlers
+            patched_client._missing = self._app.error_handler._missing
+            self._app.error_handler = patched_client
```

Comment:
I am using a custom error handler in my sanic app and when I tried with this PR it overrides it. Is it supposed to be like this?

Reply:
Nice catch. This was a small fault on my part. I will fix the way the error handler client is being patched.